### ISSUE TITLE
Tickets, Permissions, and Sharing data - A question about expectations

### ISSUE THREAD

**Issue by korydraughn** [irods/irods#6501]
## Question
It's not uncommon to give another user access to data they don't own. There are two ways to do this:
- Create a ticket for the target object and give the ticket to the user
- Or, modify the permissions on the target object and then on the parent collection

The point of interest here is that using a ticket doesn't require modifying permissions on the parent collection. Great. The user can't look around the parent collection. All they can do is access the data associated with the ticket.

Q. Should modifying the permissions on the parent collection be necessary?
Q. Should modifying permissions on the target object result in the same behavior as using a ticket?

**Comment by paulborgermans** [comment]
I think not have to modify the permissions on the parent collection would be a great change
And AFAIK a ticket is different, because this allows possibly anonymous access and more controls (like DNS based allow/deny) right?

**Comment by korydraughn** [comment]
> And AFAIK a ticket is different, because this allows possibly anonymous access and more controls (like DNS based allow/deny) right?

Yes.

**Comment by niravmerchant** [comment]
For applications that allow browsing of collections, being able to navigate the directory structure to reach the file is helpful. 
Especially when multiple shares are occurring in the same directory/location over time so you can pick and choose the files/directories needed.

**Comment by korydraughn** [comment]
That is true. And we don't want to change that. This issue is about situations where a user wants to share data objects without having to share their entire collection. The user does not want other users to be able to navigate their collections.

Tickets support this use-case. It feels more correct to only require that a user have read permission on a data object to access it. It feels incorrect to require read permission on the parent collection just to share a single data object.

Making this change to the permission model would make collaborating easier without completely removing the need for tickets.

**Comment by niravmerchant** [comment]
That is correct, if I am not mistaken the iget command with ticket access will still need full path so the collection to download/access it. (has been a while since I used it)
We give read access to collection but not to additional objects in it. So if external user navigates to collection, they cannot see additional objects as they do not have read permission to those.
________________________________
From: Kory Draughn ***@***.***>
Sent: Tuesday, August 9, 2022 5:51 AM
To: irods/irods ***@***.***>
Cc: Merchant, Nirav C - (nirav) ***@***.***>; Comment ***@***.***>
Subject: [EXT]Re: [irods/irods] Tickets, Permissions, and Sharing data - A question about expectations (Issue #6501)


External Email

That is true. And we don't want to change that. This issue is about situations where a user wants to share data objects without having to share their entire collection. The user does not want other users to be able to navigate their collections.

Tickets support this use-case. It feels more correct to only require that a user have read permission on a data object to access it. It feels incorrect to require read permission on the parent collection just to share a single data object.

Making this change to the permission model would make collaborating easier without completely removing the need for tickets.

**Comment by korydraughn** [comment]
I just confirmed what you said. Giving a user read permission on a collection does not expose the objects inside of it.

I think the question still remains.

Q. Should users be required to grant read permission on the parent collection?
Q. @niravmerchant Do you see possible issues with removing this requirement?

**Comment by korydraughn** [comment]
This change potentially shrinks the amount of information stored in the database too because we don't have to maintain a link between tables (object_id, permission_id).

**Comment by niravmerchant** [comment]
Q. Should users be required to grant read permission on the parent collection?
I vote yes
Q. @niravmerchant<https://github.com/niravmerchant> Do you see possible issues with removing this requirement?
Since most of our sharing/access is by named user and not tickets it will not break things.

________________________________
From: Kory Draughn ***@***.***>
Sent: Tuesday, August 9, 2022 6:49 AM
To: irods/irods ***@***.***>
Cc: Merchant, Nirav C - (nirav) ***@***.***>; Mention ***@***.***>
Subject: [EXT]Re: [irods/irods] Tickets, Permissions, and Sharing data - A question about expectations (Issue #6501)


External Email

I just confirmed what you said. Giving a user read permission on a collection does not expose the objects inside of it.

I think the question still remains.

Q. Should users be required to grant read permission on the parent collection?
Q. @niravmerchant<https://github.com/niravmerchant> Do you see possible issues with removing this requirement?

**Comment by niravmerchant** [comment]
If it is makes things efficient I am all for it. Please go ahead.

________________________________
From: Kory Draughn ***@***.***>
Sent: Tuesday, August 9, 2022 6:53 AM
To: irods/irods ***@***.***>
Cc: Merchant, Nirav C - (nirav) ***@***.***>; Mention ***@***.***>
Subject: [EXT]Re: [irods/irods] Tickets, Permissions, and Sharing data - A question about expectations (Issue #6501)


External Email

This change potentially shrinks the amount of information stored in the database too because we don't have to maintain a link between tables (object_id, permission_id).

**Comment by niravmerchant** [comment]
I don't think we need a hard requirement to share parent collection, this goes back to my original mention of allowing/supporting browsing style behavior to support navigation within collections 
Given the need/desire for robust S3 API compatibility, I would suggest that we should align with S3 pre-signed URL  style access to guide our choices in this regard.
